src/consensus/pbft/pbft.h

```diff
@@ -179,6 +184,19 @@ namespace pbft
         return msg->size();
       }
 
+      if (should_encrypt(msg->tag()))
+      {
+        PbftHeader hdr = {PbftMsgType::encrypted_pbft_message, id};
+
+        auto space = (size_t)msg->size();
```

Comment:
There is considerable code reuse here with line 200+

Reply:
size_t space = msg->size(); surely?